### DISCUSSION TITLE
Standardize configuration UX

### DISCUSSION THREAD

**Discussion by ReplicaObscura** [jseidl/magic-areas#241] [Ideas / Feature Requests]
I think it would be beneficial if the overall configuration workflow for MA worked more like other integrations. By this, I mean no configuration.yml entry needed and no auto-creation of integrations in the UI.

According to the HA developer docs, integrations should not create themselves, at most they should use discovery to suggest integrations. But I would prefer if MA didn't even do that, because I have a lot of areas and don't necessarily want an integration for each one of those, so they'd just be clutter.

My preferred workflow for MA would be that you just install through HACS and then start adding integrations in the UI for each area you want MA to handle.

This would be in line with other integrations, would eliminate some confusion, would make uninstalling simpler, and could maybe even help pave the way for MA not requiring restarts of HA which would be a dream come true.

 I'd be happy to work on this and submit a PR if this is a direction you're interested in going, or at least interested in seeing it in action.

**Comment by jseidl** [comment]
I'd totally be interested! The reason why it is how it is today is that this is the way I've came to do it so the meta areas would work.

If there's a more compliant way of getting those to work while keeping things standard that would be great!

**Reply by ReplicaObscura** [reply]
Are the meta areas always required, or could those also be added manually like the other areas? I'm wondering if maybe when adding a Magic Areas integration, there could just be a select list of areas which don't yet have an integration added, along with the meta areas if they haven't yet been added. Then submitting it would create the config entity and then pop up the normal UI config form for that area like it works today.

**Reply by jseidl** [reply]
I wouldn't say they're always required. They definitely a feature to keep tho. It could be added manually.

I believe having a first step that lists all the areas that don't have an Magic Area entry yet, works (plus the meta areas).

**Comment by jseidl** [comment]
> My preferred workflow for MA would be that you just install through HACS and then start adding integrations in the UI for each area you want MA to handle.

This is now the workflow for MA 3.1.0 and over.
